akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -789,10 +789,16 @@ private[akka] class ShardRegion(
   def receiveClusterEvent(evt: ClusterDomainEvent): Unit = evt match {
     case MemberUp(m) =>
       addMember(m)
+
     case MemberLeft(m) =>
-      addMember(m)
+      if (m.uniqueAddress != cluster.selfUniqueAddress && matchingCoordinatorRole(m))
```

Comment:
Maybe the `MemberLeft` case can be reverted to just `addMember`?  Perhaps it's less likely to be a win in that case?